### PR TITLE
Add "verify-messages" target

### PR DESCRIPTION
**Pull request by JavierMatosD**
This PR adds a script that ensures all the messages declared in `message-data.inc.h` are actually getting used. This also adds the check to our PR workflow.

### CODE REVIEW

**Comment by JavierMatosD** [comment]
I want to remove all the unused, but waiting on review for the script.

File: .github/workflows/pr.yaml
Comment (BillyONeal) [inline .github/workflows/pr.yaml:17]: Can you add this to format instead so we aren't stuck waiting for another machine? Real builds already take much more time anyway...

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:19]: Can you just read the json file instead of trying to parse the C++ with regex like this?

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:26]:
```suggestion
$used_messages = Get-ChildItem -Path $SEARCH_DIR -Include @('*.cpp', '*.h') -Recurse |
```

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:27]: Should this be `\bmsg[A-Za-z0-9_]+\b` ?

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:58]: Do we need this part?

**Review by BillyONeal — COMMENTED** [review]
Can you explain why we aren't finding the use inside message-data itself?

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [reply scripts/verifyMessages.ps1:58]: (To be clear, I mean the timer, not the write-hosts above)

Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:58]: oops. Sorry, I had that there because the first implementation took like 5 min so I was trying to rework it to be faster. Forgot to remove the timer. :)

File: scripts/verifyMessages.ps1
Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:19]: That's a good idea. Originally, I was looping through the `declared_messages` and searching the source code for any instance of the message. (that was a bad idea lol) so I didn't want to deal with the artifact messages. But now that I'm not doing that, I can use the json. :)

File: .github/workflows/pr.yaml
Comment (JavierMatosD) [reply .github/workflows/pr.yaml:17]: Sounds good!

**Comment by JavierMatosD** [comment]
> Can you explain why we aren't finding the use inside message-data itself?

Not sure what you mean. Can you elaborate? What aren't we finding?

File: scripts/verifyMessages.ps1
Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:16]: Never mind. The json files also have artifact messages. There's no way to distinguish messages from CPP world and from artifact world. The only way I can think of that can distinguish (without modifying the json structure) is that the artifact messages are always after the CPP messages. The CPP messages are in alphabetical order so you can guesstimate which is the first artifact message. That said, we don't enforce sorting of messages in the header file.

File: .github/workflows/pr.yaml
Comment (BillyONeal) [inline .github/workflows/pr.yaml:35]: It seems like this should reuse the configure from line 39/44 rather than configuring again?

File: CMakeLists.txt
Comment (BillyONeal) [inline CMakeLists.txt:562]: Do we need these 3 very generic variable names? I think it would be just as clear or clearer to just expand them inline.

(Note that I think `ARTIFACT_MESSAGES_FILE` is OK because it describes exactly what it is and is unlikely to ever conflict with anything else)

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:57]: Please add a \n

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:9]: Stupid question: Instead of having the caller tell us what to check, should this script use $PSScriptRoot to figure it  out?

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [inline scripts/verifyMessages.ps1:39]: If this is slow, I think it could be replaced with `Compare-Object`. (I hope `Compare-Object` doesn't use the N^2 algorithm like this...)

**Review by BillyONeal — COMMENTED** [review]
LGTM assuming the messages are actually fixed of course :)

File: scripts/verifyMessages.ps1
Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:39]: I could do that but the current implementation is not slow.

File: scripts/verifyMessages.ps1
Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:9]: Not a stupid question. That makes sense to me.

File: scripts/verifyMessages.ps1
Comment (BillyONeal) [reply scripts/verifyMessages.ps1:52]: It's still missing :(

Maybe editor is fighting you?

**Review by BillyONeal — APPROVED** [review]

File: scripts/verifyMessages.ps1
Comment (JavierMatosD) [reply scripts/verifyMessages.ps1:52]: I see the newline being added. I believe **`n** is the powershell new line

**Pull request merged**
